mono/sgen/sgen-gc.c

```diff
@@ -3123,6 +3020,12 @@ sgen_gc_init (void)
 	if (major_collector.needs_thread_pool)
 		sgen_workers_init (1);
 
+	if (!soft_limit) {
+		gint64 ram_size = mono_determine_physical_ram_size ();
+		if (ram_size > 0)
+			soft_limit = ram_size * SGEN_SOFT_HEAP_LIMIT_RATIO;
```

Comment:
The soft limit doesn't enforce a limit on heap sizes. The ideal of the 75% limit is for it to be a hard limit.

By default, the managed heap must not go above that.

This gives us the memory headroom we need to OOM.